### ISSUE TITLE
feffit with an array of uncertainty in chi(k)

### ISSUE THREAD

**Issue by bruceravel** [xraypy/xraylarch#133]
Matt,

In the doc, you say [Though usually ϵ is taken to be a scalar value, it can be specfied as an array (of the same length as χ(k)) if more accurate measures for the uncertainty of the data is available.](http://xraypy.github.io/xraylarch/xafs/feffit.html#fit-statistics-and-goodness-of-fit-meassures-for-feffit) 

I was doing some code diving, but wasn't able to figure out the syntax for this.  Could you provide an example.  Just explaining how to modify [this](https://github.com/xraypy/xraylarch/blob/master/examples/feffit/Feffit_01.lar) would be quite enough to get me started.

Thanks!
B

**Comment by newville** [comment]
@bruceravel  yeah, that seems broken.  I think the right behavior should be to allow 'epsilon_k' as a keyword arg to `feffit_dataset()`, and allow this to either be a scalar value or an array that is the same length as `k` and `chi` -- the uncertainty at each k point.  Will fix.

**Comment by newville** [comment]
@bruceravel should be fixed now.  Essentially, if a data group has `epsilon_k` as either a scalar or array the same length as `k` then that is used for the fit.   Or you can now specify `epsilon_k` in `feffit_dataset()`.

**Comment by bruceravel** [comment]
Matt, looking at the example `Feffit_01.lar`, I was surprised that the array-valued `cu.epsilon_k` only had an impact on the evaluation of chi-square.  The best fit values and uncertainties of the 4 parameters were identical to 6 digits.

Am I thinking about this wrongly?  I certainly don't expect a big change for Cu foil data. But I would have thought that, with a non-integer value for epsilon, the numerical convergence of the fit would have resulted in differences in the smaller decimal places of the parameters.

**Comment by newville** [comment]
@bruceravel Hm, yeah something seems wrong.  Will investigate.

**Comment by newville** [comment]
@bruceravel  Ah, yes, I see that `epsilon_r` is set to its mean value......
